src/main/java/com/netflix/imflibrary/st2067_2/AbstractApplicationComposition.java

```diff
@@ -149,6 +149,11 @@ public AbstractApplicationComposition(@Nonnull IMFCompositionPlaylistType imfCom
         imfErrorLogger.addAllErrors(IMFCoreConstraintsChecker.checkVirtualTracks(compositionPlaylistType, this
                 .virtualTrackMap, essenceDescriptorListMap, this.regXMLLibDictionary, homogeneitySelectionSet));
 
+        if (IMFCoreConstraintsChecker.hasIABVirtualTracks(compositionPlaylistType, virtualTrackMap)) {
```

Comment:
Is there any benefit to first checking `hasIABVirtualTracks`? 
The `checkIABVirtualTrack` method appears to correctly ignore non-IAB tracks. It seems like it would be cleaner to let it run always, and skip the tracks there.

Reply:
I find the code easier to read. You don't have to go into the checkIABVirtualTrack to see what's happening.